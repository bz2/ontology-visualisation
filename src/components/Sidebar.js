import React from 'react'
import { connect } from 'redux-zero/react'
import { useTranslation } from 'react-i18next'
import PropTypes from 'prop-types'
import {
  GiTronArrow
} from 'react-icons/gi'
import {
  GrNodes
} from 'react-icons/gr'
import {
  AiFillSetting,
} from 'react-icons/ai'
import actions from '../store/actions'
import {
  SIDEBAR_VIEW_NODES,
  SIDEBAR_VIEW_EDGES,
  SIDEBAR_VIEW_SETTINGS
} from '../constants/views'

const Sidebar = ({
  sidebarView,
  setStoreState,
}) => {
  const { t } = useTranslation()

  return (
    <aside>
      <div className="aside-bar">
        <button
          type="button"
          title={t('nodesOptions')}
          className={sidebarView === SIDEBAR_VIEW_NODES ? 'aside-bar-button-selected' : ''}
          onClick={() => setStoreState('sidebarView', SIDEBAR_VIEW_NODES)}
        >
          <GrNodes />
        </button>

        <button
          type="button"
          title={t('edgesOptions')}
          className={sidebarView === SIDEBAR_VIEW_EDGES ? 'aside-bar-button-selected' : ''}
          onClick={() => setStoreState('sidebarView', SIDEBAR_VIEW_EDGES)}
        >
          <GiTronArrow />
        </button>

        <button
          type="button"
          title={t('settings')}
          className={sidebarView === SIDEBAR_VIEW_SETTINGS ? 'aside-bar-button-selected' : ''}
          onClick={() => setStoreState('sidebarView', SIDEBAR_VIEW_SETTINGS)}
        >
          <AiFillSetting />
        </button>
      </div>

      ADD CONTENT
    </aside>
  )
}

Sidebar.propTypes = {
  sidebarView: PropTypes.string.isRequired,
  setStoreState: PropTypes.func.isRequired,
}

const mapToProps = ({
  sidebarView
}) => ({
  sidebarView
})

export default connect(
  mapToProps,
  actions
)(Sidebar)
