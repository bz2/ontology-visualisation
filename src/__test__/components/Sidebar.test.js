import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Sidebar from '../../components/Sidebar'
import {
  SIDEBAR_VIEW_GRAPHS,
  SIDEBAR_VIEW_GRAPH_OPTIONS,
  SIDEBAR_VIEW_FREE_TEXT_SEARCH,
  SIDEBAR_VIEW_EDGES_SELECTION,
  SIDEBAR_VIEW_NODES_SELECTION,
  SIDEBAR_VIEW_BOUNDING_BOX,
  SIDEBAR_VIEW_NEIGHBOURHOOD,
  SIDEBAR_VIEW_SHORTEST_PATH,
  SIDEBAR_VIEW_SETTINGS,
  SIDEBAR_VIEW_NODES_FILTER,
  SIDEBAR_VIEW_EDGES_FILTER,
  SIDEBAR_VIEW_CUSTOM_QUERY,
  SIDEBAR_VIEW_EXPORT,
  SIDEBAR_VIEW_EDIT_ONTOLOGY,
  SIDEBAR_VIEW_STYLING,
  SIDEBAR_VIEW_STRUCTURED_SEARCH,
  SIDEBAR_VIEW_ENTRY_SEARCH,
  SIDEBAR_VIEW_SYNONYMS
} from '../../constants/views'

const setup = ({
  isSidebarOpen,
  sidebarView
}) => {
  const props = {
    isSidebarOpen,
    sidebarView,
    updateStoreValue: jest.fn(),
  }

  const component = shallow(<Sidebar {...props} />)

  return {
    component,
    props
  }
}

describe('Sidebar', () => {
  it('should match snapshot when sidebar closed', () => {
    const {
      component
    } = setup({
      isSidebarOpen: false,
      sidebarView: SIDEBAR_VIEW_GRAPHS
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_ENTRY_SEARCH', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_ENTRY_SEARCH
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_GRAPHS', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_GRAPHS
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_GRAPH_OPTIONS', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_GRAPH_OPTIONS
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_FREE_TEXT_SEARCH', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_FREE_TEXT_SEARCH
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_BOUNDING_BOX', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_BOUNDING_BOX
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_NEIGHBOURHOOD', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_NEIGHBOURHOOD
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_NODES_SELECTION', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_NODES_SELECTION
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_EDGES_SELECTION', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_EDGES_SELECTION
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_SHORTEST_PATH', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_SHORTEST_PATH
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_SETTINGS', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_SETTINGS
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_NODES_FILTER', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_NODES_FILTER
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_EDGES_FILTER', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_EDGES_FILTER
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_CUSTOM_QUERY', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_CUSTOM_QUERY
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_EXPORT', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_EXPORT
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_STYLING', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_STYLING
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_EDIT_ONTOLOGY', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_EDIT_ONTOLOGY
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_STRUCTURED_SEARCH', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_STRUCTURED_SEARCH
    })

    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when sidebar open and SIDEBAR_VIEW_SYNONYMS', () => {
    const {
      component
    } = setup({
      isSidebarOpen: true,
      sidebarView: SIDEBAR_VIEW_SYNONYMS
    })

    expect(toJson(component)).toMatchSnapshot()
  })
})
