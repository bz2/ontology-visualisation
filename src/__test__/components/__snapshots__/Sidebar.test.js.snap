// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match snapshot when sidebar closed 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={false}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="networkGraphs"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_BOUNDING_BOX 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="boundingBox"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_CUSTOM_QUERY 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="customQuery"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_EDGES_FILTER 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="edgesFilter"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_EDGES_SELECTION 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="edgesSelection"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_EDIT_ONTOLOGY 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="editOntology"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_EXPORT 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="export"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_FREE_TEXT_SEARCH 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="freeTextSearch"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_GRAPHS 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="networkGraphs"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_NEIGHBOURHOOD 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="nodeNeighbourhood"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_NODES_FILTER 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="nodesFilter"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_NODES_SELECTION 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="nodesSelection"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_SETTINGS 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="settings"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_SHORTEST_PATH 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="shortestPath"
>
  <Component />
</Connect>
`;

exports[`Sidebar should match snapshot when sidebar open and SIDEBAR_VIEW_VERSIONING 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
    }
  }
  isSidebarOpen={true}
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
  sidebarView="versioning"
>
  <Component />
</Connect>
`;
